replicator/src/main/scala/com/evolutiongaming/kafka/journal/replicator/TopicCommit.scala

```diff
@@ -1,44 +1,132 @@
 package com.evolutiongaming.kafka.journal.replicator
 
 import cats.Applicative
-import cats.data.NonEmptyMap as Nem
+import cats.data.{NonEmptyMap as Nem, NonEmptySet}
+import cats.effect.*
 import cats.effect.kernel.Concurrent
-import cats.effect.{Clock, Ref}
 import cats.syntax.all.*
 import com.evolutiongaming.catshelper.ClockHelper.*
 import com.evolutiongaming.catshelper.DataHelper.*
+import com.evolutiongaming.catshelper.Log
 import com.evolutiongaming.kafka.journal.KafkaConsumer
+import com.evolutiongaming.kafka.journal.replicator.commit.AsyncPeriodicTopicCommit
 import com.evolutiongaming.kafka.journal.util.TemporalHelper.*
 import com.evolutiongaming.skafka.*
+import com.evolutiongaming.skafka.consumer.RebalanceCallback
 
 import java.time.Instant
+import scala.annotation.nowarn
 import scala.collection.immutable.SortedMap
 import scala.concurrent.duration.*
 
-trait TopicCommit[F[_]] {
+/**
+ * Kafka partition offset commit logic for replicator internal topic processing API
+ * ([[TopicConsumer]], [[ConsumeTopic]]).
+ *
+ * @tparam F effect type
+ */
+private[journal] trait TopicCommit[F[_]] {
+// journal-private because the places where it is used are journal-private - TopicConsumer, ConsumeTopic
 
+  /**
+   * Mark offsets for commit.
+   * 
+   * The method should be called after each processing step, i.e., after each consumer poll result processing.
+   * 
+   * Please note that offsets passed here should follow the same logic as for Java consumer commit methods:
+   * if offset N is known to be processed, offset N+1 should be committed - offset from which you want to start your
+   * processing next time.
+   * 
+   * Depending on the underlying implementation, this method might execute commit synchronously with the call or
+   * schedule it for later.
+   * 
+   * @param offsets partition offsets to commit
+   */
   def apply(offsets: Nem[Partition, Offset]): F[Unit]
+
+  /**
+   * Notify topic commit logic about new partitions assigned during rebalance - called in
+   * [[com.evolutiongaming.skafka.consumer.RebalanceListener1.onPartitionsAssigned]].
+   * 
+   * @param partitions assigned partitions
+   *                   
+   * @return [[RebalanceCallback]]
+   */
+  def onPartitionsAssigned(@nowarn("cat=unused") partitions: NonEmptySet[Partition]): RebalanceCallback[F, Unit] =
+    RebalanceCallback.pure(())
```

Comment:
maybe to make it more expressive 
```suggestion
    RebalanceCallback.empty
```

Reply:
I was looking for smth like `.unit` but this works as well. Will be fixed